lib/vanagon/utilities/extra_files_signer.rb

```diff
@@ -0,0 +1,34 @@
+class Vanagon
+  module Utilities
+    module ExtraFilesSigner
+      class << self
+        def commands(project, mktemp, source_dir) # rubocop:disable Metrics/AbcSize
+          tempdir = nil
+          commands = []
+          # Skip signing extra files if logging into the signing_host fails
+          # This enables things like CI being able to sign the additional files,
+          # but locally triggered builds by developers who don't have access to
+          # the signing host just print a message and skip the signing.
+          Vanagon::Utilities.retry_with_timeout(3, 5) do
+            tempdir = Vanagon::Utilities::remote_ssh_command("#{project.signing_username}@#{project.signing_hostname}", "#{mktemp} 2>/dev/null", return_command_output: true)
+          end
+
+          project.extra_files_to_sign.each do |file|
+            file_location = File.join(tempdir, File.basename(file))
+            commands += [
+              "rsync -e '#{Vanagon::Utilities.ssh_command}' -rHlv --no-perms --no-owner --no-group #{File.join('$(tempdir)', source_dir, file)} #{project.signing_username}@#{project.signing_hostname}:#{tempdir}",
```

Comment:
(Minor) style request: Use long options as much as possible/compatible; also consider pre-computing paths outside for clarity.

Something like:

```ruby

local_source_path = File.join('$(tempdir)', source_dir, file)
remote_destination_path = "#{project.signing_username}@#{project.signing_hostname}:#{tempdir}"

commands += [
  "rsync --rsh=#{Vanagon::Utilities.ssh_command} --verbose --recursive --hard-links --links  --no-perms --no-owner --no-group #{local_source_path} #{remote_destination_path}"
...
```